src/Utilities/Compiler/Extensions/IOperationExtensions.cs

```diff
@@ -801,6 +801,331 @@ public static bool IsSetMethodInvocation(this IPropertyReferenceOperation operat
 
             return parameterOrderedArguments.ToImmutableArray();
         }
+
+        // Copied from roslyn https://github.com/dotnet/roslyn/blob/master/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs#L25
+
+#if CODEANALYSIS_V3_OR_BETTER
```

Comment:
Some operations used here newer or so not recognised for **BannedAPiAnalyzers**:
```
D:\dotnet\roslyn-analyzers\src\Utilities\Compiler\Extensions\IOperationExtensions.cs(854,26): error CS0246: The type or namespace name 'IRecursivePatternOperation' could not be found (are you missing a using directive or an assembly reference?) [D:\dotnet\roslyn-analyzers\src\Microsoft.CodeAnalysis.BannedApiAnalyzers\Core\Microsoft.CodeAnalysis.BannedApiAnalyzers.csproj]
D:\dotnet\roslyn-analyzers\src\Utilities\Compiler\Extensions\IOperationExtensions.cs(865,26): error CS0246: The type or namespace name 'ISwitchExpressionArmOperation' could not be found (are you missing a using directive or an assembly reference?) [D:\dotnet\roslyn-analyzers\src\Microsoft.CodeAnalysis.BannedApiAnalyzers\Core\Microsoft.CodeAnalysis.BannedApiAnalyzers.csproj]
D:\dotnet\roslyn-analyzers\src\Utilities\Compiler\Extensions\IOperationExtensions.cs(883,26): error CS0246: The type or namespace name 'IPropertySubpatternOperation' could not be found (are you missing a using directive or an assembly reference?) [D:\dotnet\roslyn-analyzers\src\Microsoft.CodeAnalysis.BannedApiAnalyzers\Core\Microsoft.CodeAnalysis.BannedApiAnalyzers.csproj]
D:\dotnet\roslyn-analyzers\src\Utilities\Compiler\Extensions\IOperationExtensions.cs(962,42): error CS0246: The type or namespace name 'IReDimClauseOperation' could not be found (are you missing a using directive or an assembly reference?) [D:\dotnet\roslyn-analyzers\src\Microsoft.CodeAnalysis.BannedApiAnalyzers\Core\Microsoft.CodeAnalysis.BannedApiAnalyzers.csproj]
D:\dotnet\roslyn-analyzers\src\Utilities\Compiler\Extensions\IOperationExtensions.cs(965,56): error CS0246: The type or namespace name 'IReDimOperation' could not be found (are you missing a using directive or an assembly reference?) [D:\dotnet\roslyn-analyzers\src\Microsoft.CodeAnalysis.BannedApiAnalyzers\Core\Microsoft.CodeAnalysis.BannedApiAnalyzers.csproj]
D:\dotnet\roslyn-analyzers\src\Utilities\Compiler\Extensions\IOperationExtensions.cs(1062,22): error CS0246: The type or namespace name 'ICoalesceAssignmentOperation' could not be found (are you missing a using directive or an assembly reference?) [D:\dotnet\roslyn-analyzers\src\Microsoft.CodeAnalysis.BannedApiAnalyzers\Core\Microsoft.CodeAnalysis.BannedApiAnalyzers.csproj]
D:\dotnet\roslyn-analyzers\src\Utilities\Compiler\Extensions\IOperationExtensions.cs(854,26): error CS0246: The type or namespace name 'IRecursivePatternOperation' could not be found (are you missing a using directive or an assembly reference?) [D:\dotnet\roslyn-analyzers\src\PerformanceSensitiveAnalyzers\Core\Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.csproj]
D:\dotnet\roslyn-analyzers\src\Utilities\Compiler\Extensions\IOperationExtensions.cs(865,26): error CS0246: The type or namespace name 'ISwitchExpressionArmOperation' could not be found (are you missing a using directive or an assembly reference?) [D:\dotnet\roslyn-analyzers\src\PerformanceSensitiveAnalyzers\Core\Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.csproj]
D:\dotnet\roslyn-analyzers\src\Utilities\Compiler\Extensions\IOperationExtensions.cs(883,26): error CS0246: The type or namespace name 'IPropertySubpatternOperation' could not be found (are you missing a using directive or an assembly reference?) [D:\dotnet\roslyn-analyzers\src\PerformanceSensitiveAnalyzers\Core\Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.csproj]
D:\dotnet\roslyn-analyzers\src\Utilities\Compiler\Extensions\IOperationExtensions.cs(962,42): error CS0246: The type or namespace name 'IReDimClauseOperation' could not be found (are you missing a using directive or an assembly reference?) [D:\dotnet\roslyn-analyzers\src\PerformanceSensitiveAnalyzers\Core\Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.csproj]
D:\dotnet\roslyn-analyzers\src\Utilities\Compiler\Extensions\IOperationExtensions.cs(965,56): error CS0246: The type or namespace name 'IReDimOperation' could not be found (are you missing a using directive or an assembly reference?) [D:\dotnet\roslyn-analyzers\src\PerformanceSensitiveAnalyzers\Core\Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.csproj]
D:\dotnet\roslyn-analyzers\src\Utilities\Compiler\Extensions\IOperationExtensions.cs(1062,22): error CS0246: The type or namespace name 'ICoalesceAssignmentOperation' could not be found (are you missing a using directive or an assembly reference?) [D:\dotnet\roslyn-analyzers\src\PerformanceSensitiveAnalyzers\Core\Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.csproj]
```

Reply:
Great, thanks!